addons/web/static/src/core/domain_selector/domain_selector.js

```diff
@@ -98,6 +102,13 @@ export class DomainSelector extends Component {
         ]);
 
         const pathsInfo = await this.loadPathsInfo(p.resModel, paths);
+        if (p.readonly) {
+            const pathFieldDefs = Object.fromEntries(
+                Object.entries(pathsInfo).map(([path, pathInfo]) => [path, pathInfo.fieldDef])
+            );
+            const idsByModel = extractIdsFromDomain(domain, pathFieldDefs);
+            this.displayNames = await this.loadDisplayNames(idsByModel);
```

Comment:
we could maybe always fetch the display names at this level and let the autocomplete components find them in the name service cache